src/index.js

```diff
@@ -1,46 +1,91 @@
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'finish Microverse Project',
-    completed: false,
-    index: 3,
-  },
-];
-
-const saveTaskToLocal = (value) => {
+// eslint-disable-next-line
+import { changeState } from './events.js';
+
+export const listContainer = document.querySelector('.list');
+// eslint-disable-next-line
+let tasks = [];
+export { tasks };
+
+export const saveTaskToLocal = (value) => {
   localStorage.setItem('task', JSON.stringify(value));
 };
 
-saveTaskToLocal(tasks);
+const getTaskFromLocal = () => {
+  const dataFromLocal = JSON.parse(localStorage.getItem('task'));
+  if (dataFromLocal) {
+    _.forEach(dataFromLocal, (data, i) => {
+      data.index = i + 1;
+      tasks.push(data);
+    });
+  }
+};
 
-const getTaskFromLocal = () => JSON.parse(localStorage.getItem('task'));
+getTaskFromLocal();
 
-const populateList = (values) => {
-  values.forEach((toDo, i) => {
+const updateTasks = (data) => {
+  tasks = data;
+};
+
+export const removeAll = () => {
+  updateTasks(tasks.filter((task) => task.completed === false));
+  tasks.forEach((task, i) => {
+    task.index = i + 1;
+  });
+  saveTaskToLocal(tasks);
+};
+
+export const populateList = (values) => {
+  const sortedTasks = _.sortBy(values, 'index');
+  _.forEach(sortedTasks, (toDo) => {
     const htmlText = `
-      <li class='item'>     
-        <input type='checkbox' class='${toDo.completed} checkbox' id='${toDo.description[0]}${toDo.index}' />
-        <label for='${toDo.description[0]}${toDo.index}' class='item-description'>${toDo.description}</label> 
-        <ion-icon name='ellipsis-vertical-outline' class='dynamic-icons'></ion-icon>
-      </li>`;
+    <li class='item'>
+      <input type='checkbox' class='checkbox' id='${toDo.description[0]}${
+  toDo.index
+}' ${toDo.completed ? 'checked' : ''}/>
+      <div class="inside-div">
+        <span contentEditable='true' class='item-description ${
+  toDo.completed ? 'item-description-done' : ''
+}'>${
+  toDo.description
+}<ion-icon name="trash-outline" class="display-icon trash-icon"></ion-icon></span>
+        <ion-icon name='ellipsis-vertical-outline' class='dynamic-icons'></ion-icon>    
+      </div>
+    </li>`;
 
-    if (i + 1 === toDo.index) listContainer.innerHTML += htmlText;
+    listContainer.insertAdjacentHTML('beforeend', htmlText);
   });
 };
 
-populateList(getTaskFromLocal());
-changeState(getTaskFromLocal());
\ No newline at end of file
+populateList(tasks);
+changeState(tasks);
+
+const taskIcon = Array.from(document.querySelectorAll('.dynamic-icons'));
+const trashIcons = [...document.querySelectorAll('.trash-icon')];
+const textDescription = [...document.querySelectorAll('.item-description')];
+const liItem = [...document.querySelectorAll('.item')];
+
+_.forEach(liItem, (item, i) => {
+  textDescription[i].addEventListener('focus', () => {
+    item.style.backgroundColor = 'rgba(248, 244, 2, 0.322)';
+    taskIcon[i].classList.add('display-icon');
```

Comment:
- [ ] Editing task descriptions is not working as expected, the new task description is not persisted after the edit. Let's fix that

Reply:
Is working, You only have to click on the task , write something or modify the taks and after that press TAB key or just click in other side, out of the task.